tests/UnifiedSpecTests/ExpectedError.php

```diff
@@ -166,8 +170,15 @@ public function assert(?Throwable $e = null): void
         }
 
         if (isset($this->matchesResultDocument)) {
-            assertInstanceOf(CommandException::class, $e);
-            assertThat($e->getResultDocument(), $this->matchesResultDocument, 'CommandException result document matches');
+            assertThat($e, logicalOr(isInstanceOf(CommandException::class), isInstanceOf(WriteException::class)));
+
+            if ($e instanceof CommandException) {
+                assertThat($e->getResultDocument(), $this->matchesResultDocument, 'CommandException result document matches');
+            } elseif ($e instanceof WriteException) {
+                $writeErrors = $e->getWriteResult()->getErrorReplies();
+                assertCount(1, $writeErrors);
+                assertThat($writeErrors[0], $this->matchesResultDocument, 'WriteException result document matches');
```

Comment:
This isn't addressed in the spec (see: [expectedError](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#expectederror)) but this is the correct approach. I intentionally only issued to single commands in the corresponding write command tests for this reason.